plugins/nexus-repository-manager/src/hooks/useNexusRepositoryManagerAppData.ts

```diff
@@ -0,0 +1,64 @@
+import { type Entity } from '@backstage/catalog-model';
+
+import { NEXUS_REPOSITORY_MANAGER_CONFIG_ANNOTATIONS } from '../annotations';
+import { Annotation, type SearchServiceQuery } from '../types';
+
+type UseNexusRepositoryManagerAppDataArgs = {
+  entity: Entity;
+  ANNOTATIONS: Readonly<Annotation[]>;
+};
+
+export const useNexusRepositoryManagerAppData = ({
+  entity,
+  ANNOTATIONS,
+}: UseNexusRepositoryManagerAppDataArgs): {
+  title: string;
+  query: SearchServiceQuery;
+} => {
+  const value = ANNOTATIONS.reduce(
+    (acc, v) => {
+      const repository = entity?.metadata.annotations?.[v.annotation];
+
+      if (!repository) {
+        return acc;
+      }
+
+      acc.repositories.push(repository);
+      const query = v.query
+        ? Object.assign(acc.query, v.query(repository))
+        : acc.query;
+
+      return {
+        ...acc,
+        query,
+      };
+    },
+    { repositories: [], query: {} } as {
+      repositories: string[];
+      query: SearchServiceQuery;
+    },
+  );
+
+  if (value.repositories.length === 0) {
+    throw new Error(`A Nexus Repository Manager annotation could not be found`);
+  }
+
+  let title: string | undefined;
+  NEXUS_REPOSITORY_MANAGER_CONFIG_ANNOTATIONS.forEach(v => {
```

Comment:
`NEXUS_REPOSITORY_MANAGER_CONFIG_ANNOTATIONS` is const right which has one item in the list so is there a need to perform forEach and switch for one case?

Reply:
This was done to make sure we handle each config annotation in case we add more.